plugins/module_utils/user.py

```diff
@@ -359,9 +359,18 @@ def user_mod(cursor, user, host, host_all, password, encrypted,
                     revoke_privs = list(set(new_priv[db_table]) & set(curr_priv[db_table]))
                 else:
                     # When replacing (neither append_privs nor subtract_privs), grant all missing privileges
-                    # and revoke existing privileges that were not requested.
+                    # and revoke existing privileges that were not requested...
                     grant_privs = list(set(new_priv[db_table]) - set(curr_priv[db_table]))
                     revoke_privs = list(set(curr_priv[db_table]) - set(new_priv[db_table]))
+
+                    # ... but do not revoke GRANT option when it's already allowed
+                    # and already in privs.
+                    #
+                    # For more details
+                    # https://github.com/ansible-collections/community.mysql/issues/77#issuecomment-1209693807
+                    if 'GRANT' in new_priv[db_table] and 'GRANT' in curr_priv[db_table]:
+                        grant_privs.append('GRANT')
```

Comment:
```suggestion
                    if 'GRANT' in grant_privs:
                        grant_option = False  # grant option is wanted, don't revoke it
```
This should solve the issue.

Reply:
Oh, my suggestion would not work if multiple tables with varying grant-option situations are processed. Instead, this should work (also not tested):
```python
# only revoke grant option if it exists and absence is requested
grant_option = 'GRANT' in revoke_privs and 'GRANT' not in grant_privs
```

Reply:
I think that  `and 'GRANT' not in grant_privs` will always be `False`.

Reply:
I'm not sure to undersand.

Did you mean to replace:

```
                    if 'GRANT' in new_priv[db_table] and 'GRANT' in curr_priv[db_table]:
                        grant_privs.append('GRANT')
```

by

```
                    # only revoke grant option if it exists and absence is requested
                    grant_option = 'GRANT' in revoke_privs and 'GRANT' not in grant_privs
```

????

Reply:
@rsicart Yes, I meant to replace the whole if statement with the last suggestion, like it is implemented now.